src/main/java/org/myrobotlab/codec/CodecUtils.java

```diff
@@ -270,20 +275,50 @@ public static String capitalize(final String line) {
      * using the selected JSON backend.
      *
      * @param json  The JSON to be deserialized in String form
-     * @param clazz The target class.
+     * @param clazz The target class. If a class is not supplied the default class returned will be an LinkedHashMap
      * @param <T>   The type of the target class.
      * @return An object of the specified class (or a subclass of) with the state
      * given by the json. Null is an allowed return object.
      * @throws JsonDeserializationException if an error during deserialization occurs.
      * @see #USING_GSON
      */
+    @SuppressWarnings("unchecked")
     public static <T> /*@Nullable*/ T fromJson(/*@Nonnull*/ String json, /*@Nonnull*/ Class<T> clazz) {
         try {
             if (USING_GSON) {
+                if (clazz == null) {
+                  clazz = (Class<T>)LinkedHashMap.class;
+                }
                 return gson.fromJson(json, clazz);
+            } else {
+              if (clazz == null) {
+ 
+                JsonFactory factory = new JsonFactory();

```

Comment:
Everything inside of the `if (clazz == null)` should be able to be replaced with `return mapper.readValue(json, (Class<T>) Object.class);`, Jackson is smart enough to handle all of that logic for you. If you want to force it to ignore the polymorphic extensions (so it guarantees a Map when the input is an object even if the class field is there) we should probably add a secondary mapper without those extensions and use that

Reply:
checking...

Reply:
I tried what you suggested - interestingly, it passed for all the primitive types ... 
But it threw ... 
```java
    // default serialization
    @SuppressWarnings({ "unchecked", "rawtypes" })
    Map<String, Object> defaultObj = (Map) CodecUtils.fromJson(json);
    assertEquals("en", defaultObj.get("language"));
```  in CodecUtilsTest.java
Because it decoded it as a Locale :thinking: it has state information ?

Reply:
Yeah that's what the virtual class field is doing. If we want to force it to deserialize objects to maps we'd need to have a second mapper without the polymorphic extensions. I can do that tomorrow if it's needed

Reply:
Our interface knows explicitly if a class is supplied
if its not supplied - Map<String, Object> works pretty well for json

That's how I kinda feel about it :shrug: what do you think ?

Reply:
Yeah it makes sense. The question is, do we want the object values to be deserialized as maps as well or keep polymorphic capabilities so they're deserialized correctly? In the latter case we'd use part of your type checking logic to check for whether json is an object and if so pass a map, otherwise use Object.class

Reply:
I don't know ... heh ... there are pros & cons regarding typing ...
I find it really amusing that everyone was excited with Python and js being typeless ...
And now quickly its "in fashion" to start annotating everything with types :roll_eyes:  ..
not sure, but I don't think i want to add anymore to this pr